ui/apps/platform/src/hooks/useFetchNamespaceDeployments.ts

```diff
@@ -1,68 +1,101 @@
+    });
+    return namespacesWithDeployments;
+}
 
 function useFetchNamespaceDeployments(selectedNamespaceIds: string[]) {
-    const [availableDeployments, setAvailableNamespaces] = useState<NamespaceWithDeployments[]>([]);
-
-    const searchClause = { 'Namespace ID': selectedNamespaceIds };
-    // If the selectedNamespaceId has not been set yet, do not run the gql query
-    const queryOptions =
-        selectedNamespaceIds.length > 0
-            ? { variables: { query: queryService.objectToWhereClause(searchClause) } }
-            : { skip: true };
-
-    const { loading, error, data } = useQuery<DeploymentResponse, { query: string }>(
-        DEPLOYMENTS_FOR_NAMESPACE_QUERY,
-        queryOptions
-    );
-
-    useEffect(() => {
-        if (!data || !data.results) {
-            setAvailableNamespaces([]);
+    const [deploymentResponse, setDeploymentResponse] = useState<ListDeploymentResponse>({
+        loading: false,
+        error: '',
+        deploymentsByNamespace: [],
+    });
+    useDeepCompareEffect(() => {
+        if (selectedNamespaceIds.length <= 0) {
+            setDeploymentResponse({
+                loading: false,
+                error: '',
+                deploymentsByNamespace: [],
+            });
         } else {
-            setAvailableNamespaces(data.results);
-        }
-        // clean up state on unmount
-        return () => setAvailableNamespaces([]);
-    }, [data]);
+            setDeploymentResponse({
+                loading: true,
+                error: '',
+                deploymentsByNamespace: [],
+            });
+            const searchQuery: Record<string, string[]> = {
+                'Namespace ID': selectedNamespaceIds,
+            };
+            const sortOption = { field: 'Deployment', reversed: 'false' };
+            listDeployments(searchQuery, sortOption, 0, 0)
+                .then((response) => {
+                    const namespacesWithDeployments: NamespaceWithDeployments[] =
+                        getNamespacesWithDeployments(response);
+                    setDeploymentResponse({
+                        loading: false,
+                        error: '',
+                        deploymentsByNamespace: namespacesWithDeployments,
+                    });
+                })
+                .catch((error) => {
+                    const message = getAxiosErrorMessage(error);
+                    const errorMessage =
+                        message ||
+                        'An unknown error occurred while getting the list of deployments';
 
-    return {
-        loading,
-        error,
-        deploymentsByNamespace: availableDeployments,
-    };
+                    setDeploymentResponse({
+                        loading: false,
+                        error: errorMessage,
+                        deploymentsByNamespace: [],
+                    });
+                });
+        }
+    }, [selectedNamespaceIds]);
```

Comment:
The effect may have to adjust for state cleanup on unmount as well
https://github.com/stackrox/stackrox/commit/764de7a19f46bfbd94764f9859a4c3f7dfc136e2#diff-6e2196a41b13bd09dbca4f6ac768c2685414ea0e02e0226e9ba0f4e90ce76e17

Reply:
That sounds like a good idea. Dave Vail is a good source of advice and review for state cleanup.